lib/src/front_end/delimited_list_builder.dart

```diff
@@ -444,33 +444,15 @@ class DelimitedListBuilder {
     var candidateIndex = _candidateBlockArgument(arguments);
     if (candidateIndex == -1) return;
 
+    // The block argument must be positional.
+    if (arguments[candidateIndex] is NamedExpression) return;
```

Comment:
Do you have an example of the undesired formatting that would result if we omit this line?

Reply:
Before this PR, you can have output like:

```dart
Stack(children: [
  widget,
  widget,
]);
```

With this change, you get:

```dart
Stack(
  children: [
    widget,
    widget,
  ],
);
```

If the argument is positional, it can still be block formatted, so this is still fine:

```dart
setState(() {
  ...
});
```

In practice, it's pretty rare for a functional call to have only one named argument that could be block formatted, so it doesn't make a big difference in practice. What it does avoid is diff churn if a function call goes from one to multiple named arguments and then is no longer able to block format what used to be the only argument (#1552).

Another way to think about it is that it using named arguments means you are getting tall style formatting (or all inline if they fit).